include/benchmark/benchmark_api.h

```diff
@@ -227,6 +231,126 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp const& value) {
 // FIXME Add ClobberMemory() for non-gnu compilers
 #endif
 
+
+
+class Counter {
+public:
+
+  typedef enum {
+    kDefaults       = 0x00,
+    /** Mark the counter as a rate. It will be presented divided by the duration of the benchmark. */
+    kIsRate         = 0x01 << 0,
+    /** Mark the counter as a thread-average quantity. It will be presented divided by the number of threads. */
+    kAvgThreads     = 0x01 << 1
+  } Flags_e;
+
+public:
+
+  Counter();
+  Counter(const char *name);
+  Counter(const char *name, double v);
+  Counter(const char *name, double v, uint32_t flags);
+  ~Counter();
+  Counter(Counter const& that);
+  Counter& operator= (Counter const& that);
+#ifdef BENCHMARK_HAS_CXX11
+  Counter(Counter && that);
+  Counter& operator= (Counter && that);
+#endif
+
+  void Set(double v) { value_ = v; }
+
+  void Finish(double cpu_time, double num_threads);
+
+  const char* Name() const { return name_; }
+  double Value() const { return value_; }
+
+  bool SameName(const char *n) const;
+
+  Counter& operator+= (double v) { value_ += v; return *this; }
+  Counter& operator-= (double v) { value_ -= v; return *this; }
+  Counter& operator*= (double v) { value_ *= v; return *this; }
+  Counter& operator/= (double v) { value_ /= v; return *this; }
+
+private:
+
+  // using raw C strings here to avoid inclusion of std::string in the public api
+  char name_buf_[16]; // strings smaller than 15 characters are stored here
+  char* name_mem_;    // whereas longer strings are stored here
+  size_t mem_size_;   // current size of name_mem_
+  char* name_;        // this points to the right one
+
+  double   value_;
+  uint32_t flags_;
+
+  void _SetName(const char *name);
```

Comment:
`_SetName` is a reserved identifier, please choose a different name.

For reference the following names are reserved and cannot be used:
* identifiers with a double underscore anywhere are reserved;
* identifiers that begin with an underscore followed by an uppercase letter are reserved;
* identifiers that begin with an underscore are reserved in the global namespace.